pkg/morserino/channels.go

```diff
@@ -22,30 +22,25 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 */
 
-import (
-	"fmt"
-	"strings"
-)
+// Structure containing all the channels used by the application
+type MorserinoChannels struct {
+	// Channel for the data flow from serial port to display goroutines
+	MessageBuffer chan string
 
-//FIXME: Add comment
-type ConsoleDisplay struct {
-	currentLine string
-	newLine     string
-}
+	// Channel indicating that all data has been displayed and that the
+	// application can be closed
+	DisplayCompleted chan bool
 
-func (cd *ConsoleDisplay) String() string {
-	//FIXME: add something useful here
-	return ""
-}
+	Done chan bool
 
-func (cd *ConsoleDisplay) Add(buff string) {
-	if strings.Contains(buff, "=") {
-		//FIXME: is the buffer one char long
-		fmt.Println("=")
-	} else {
-		fmt.Printf("%s", buff)
-	}
+	// Channel used to report back errors in goroutines
+	Error chan error
 }
 
-//TODO: add break on column
-//TODO: Add tests
+//Initialize the channels
+func (mc *MorserinoChannels) Init() {
+	mc.MessageBuffer = make(chan string, 10)
+	mc.DisplayCompleted = make(chan bool)
+	mc.Done = make(chan bool)
+	mc.Error = make(chan error, 10)
+}
```

Comment:
Nit: Prefer using Initialization function instead of `Init` method: doing the later exposes you to the risk of manipullating sturctures in an ill state (and it makes  your structure more complicated to manipulate from the caller).

```golang
var c MorserinoChannels
// c is in an ill state, if I call anything on it it will panic
c.Init()
// now we're good
```

VS 

```golang
cs := NewMorserinoChannels()
// And we're good :tada:, no risk of panics here.
```